### PR TITLE
Add relevant links to breach resolutions

### PR DESCRIPTION
**Pull request by Vinnl**
# References: 
Jira: MNTOR-1322
Figma: N/A

# Description

Some of the breach resolutions reference external services; this PR turns those into links. Also tagging @mansaj, since he added the original `<a>` tags to a couple of strings, just in case he had a plan for turning those into links that I missed.

Not all resolutions apply to all countries. I've added some code to detect the user's country through their IP address based on [similar code in Relay](https://github.com/mozilla/fx-private-relay/blob/0d14ccf6f894b09b0ae13026a4b54226ac12c025/privaterelay/utils.py#L20-L27), which I think is also applicable to Monitor. Additionally, I added a guard that checks that we do not show resolutions that involve going to the breached company's website when we don't actually know where that website is.

The way I implemented the links is by creating separate strings with the link contents, and then injected those into the resolution strings using variables. Not sure if that's the best way, but it does work /cc @flodolo.

# Screenshot (if applicable)

[Screencast from 2023-03-13 15-38-32.webm](https://user-images.githubusercontent.com/4251/224734373-0755d1f6-3b8b-4126-be0d-9c1992b6b39f.webm)

# How to test

View a breach resolution that refers to other website, and verify that it links to those websites. Additionally, the SSN and phone breach resolutions shouldn't show up outside of the US and Canada (you can set this by settings your Accept-Language), and password and security question resolutions shouldn't show up for sites that we do not have a Domain for.

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl:147]: Yes, because the previous version did not actually insert a link, just the `<a>`. Unfortunately, with the current setup of Monitor, AFAIK we're unable to dynamically update the tags like we can do in Relay (which uses `@fluent/react`), for example.

**Review by mansaj — COMMENTED** [review]
LGTM! Awesome. Pending other review comments..
Since this is pretty fluent heavy @toufali you want to take a look at it?

File: src/utils/breach-resolution.js
Comment (flodolo) [reply src/utils/breach-resolution.js:37]: This is a requirement independently of Fluent, see https://mozilla-l10n.github.io/documentation/localization/making_string_changes.html

Comment (mansaj) [reply src/utils/breach-resolution.js:37]: @flodolo Thanks for the explanation!

File: src/utils/country-code.js
Comment (flozia) [inline src/utils/country-code.js:6]: Nice!

File: src/utils/country-code.test.js
Comment (flozia) [inline src/utils/country-code.test.js:1]: Praise: … for adding thorough tests.

**Review by flozia — APPROVED** [review]

File: locales/en/breaches.ftl
Comment (toufali) [reply locales/en/breaches.ftl:147]: I've already implemented the alternative you suggested here:
https://github.com/mozilla/blurts-server/blob/df622d246bcb2b2212e3de22b016ae1a368cc1ea/src/views/partials/add-email.js#L23-L26
with corresponding translation string here:
https://github.com/mozilla/blurts-server/blob/df622d246bcb2b2212e3de22b016ae1a368cc1ea/locales/en/add-email.ftl#L21

Note the example is more involved due to multiple variables in the string, but if this pattern is preferable to localizers we could try using it instead?

File: src/utils/breach-resolution.js
Comment (toufali) [inline src/utils/breach-resolution.js:125]: Possibly more of a PM question, but should we be concerned about maintaining these specific external links?  Do we have a good means of detecting whether any of them change?

File: src/utils/breach-resolution.js
Comment (toufali) [inline src/utils/breach-resolution.js:157]: PM related question: Rather than hide the entire recommendation for specific links that may not be applicable, would there be less complexity and more content if the copy was simply changed to account for cases where a link doesn't apply?

For example, simply appending "or similar" to the following:
> Protect your email with an email masking service like { $firefoxRelayLink }

I suppose "like" above already suggests a user might use an alternative.  My point is the user might consider an email masking service regardless if it's Relay or other – it's still a good recommendation?

Additionally, I recall some logic @mansaj might have added that detects if no specific recommendation is available and then defaults to a general recommendation.  I haven't seen or tested if this PR accounts for that (might be non-issue), but my concern would be of that general default logic not firing after we remove a recommendation due to locale, and then the user is left with zero recommendations.

**Review by toufali — COMMENTED** [review]
This was quite involved indeed, nicely done!  No blockers, but questions that PM might weigh in on if you haven't checked with them already.

Geo module is great and we should keep it regardless of PM response on my question RE hiding recommendations.  Thanks!

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl:66]: Yeah, that's a good question - @flodolo, any preference?

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl:147]: Yeah, I also just realised that it would involve a larger refactoring here, since we're currently setting the same variables for every resolution string. Although with multiple links in a single string, we'd probably want to use different tags anyway, e.g. `<experian-link>Experian</experian-link>`. But since we'd just be doing a search&replace rather, I feel like that could be somewhat brittle, so I'd lean towards the current approach.

File: locales/en/breaches.ftl
Comment (flodolo) [reply locales/en/breaches.ftl:66]: My approach is that we should use terms for our own brand names, not for external companies.

In this specific case, I would go as far as not exposing them for localization at all, and hard-coding them in the link.

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: a4bbd330

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: Oh! I wasn't aware that there was a separate section for v2. I was already confused by this weird brand name tag. 333fad76

File: src/utils/country-code.js
Comment (Vinnl) [reply src/utils/country-code.js:1]: So as I understand it, that fallback should already only happen on local/Heroku: GCP should always provide an `X-Client-Region`. At least, I think it does on Relay, where this same logic is applied.

That said, _if_ it were to not be available, I'd expect the `Accept-Language` to still be more accurate than defaulting to `us`. Specifically, I expect the vast majority of people who have an `Accept-Language` with a country other than US to not actually be in the US, and most likely to be in the country that is specified there.

File: src/utils/breach-resolution.js
Comment (Vinnl) [reply src/utils/breach-resolution.js:123]: Called that out on the doc; I'm indeed not sure how faithful these companies are [to TimBL](https://www.w3.org/Provider/Style/URI.html) :P

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: Ah, I figured maybe names might be inflected or something in different languages. Hard-coded in dce383d7.

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: As discussed on Slack, removed the links in 3a5bf5ac.

File: locales/en/breaches.ftl
Comment (Vinnl) [reply locales/en/breaches.ftl]: See https://github.com/mozilla/blurts-server/pull/2900#discussion_r1135799859.

File: locales/en/breaches.ftl
Comment (flodolo) [inline locales/en/breaches.ftl:90]: This comment is out of date.

File: locales/en/breaches.ftl
Comment (flodolo) [reply locales/en/breaches.ftl:144]: The spacing hasn't been fixed here.

**Review by flodolo — COMMENTED** [review]
Please wait for review before merging pull requests with strings.

**Pull request merged**
